### PR TITLE
Fix linkage of library problem

### PR DESCRIPTION
**Pull request by sava-cska**

### CODE REVIEW

File: integration-tests/c-example/CMakeLists.txt [hunk 1]
```diff
@@ -11,4 +11,6 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 set(CMAKE_EXPORT_LINK_COMMANDS ON)
 
 add_subdirectory(lib)
-add_subdirectory(src)
\ No newline at end of file
+add_subdirectory(src)
+add_subdirectory(lib1)
```
Comment (ladisgin) [inline integration-tests/c-example/CMakeLists.txt:15]: c-example project that we can show for other people, can we avoid complication of it's structure?

File: server/src/building/Linker.cpp [hunk 2]
```diff
@@ -497,7 +497,19 @@ Result<CollectionUtils::FileSet> Linker::generateStubsMakefile(
     return stubsSet;
 }
 
-Result<utbot::Void> Linker::linkWithStubsIfNeeded(const fs::path &linkMakefile) const {
+Result<utbot::Void> Linker::linkWithStubsIfNeeded(const fs::path &linkMakefile, const fs::path &targetBitcode) const {
+    //We already have .bc file for target. If we don't remove this file, Makefile won't execute target "all",
+    //because neither stub files, nor .bc change. However, current .bc file is incorrect, because it has compiled without stubs,
+    //so it has external functions without body.
+    auto removeCommand = ShellExecTask::getShellCommandTask("rm", {"-f", targetBitcode});
```
Comment (ladisgin) [inline server/src/building/Linker.cpp:504]: Use remove from FileSystemPath.h

File: server/test/framework/Library_Test.cpp [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            auto testGen = FunctionTestGen(*request, writer.get(), TESTMODE);
+            testGen.setTargetForSource(pathToFile);
+            Status status =
+                Server::TestsGenServiceImpl::ProcessBaseTestRequest(testGen, writer.get());
+            return { testGen, status };
+        }
+    };
+
+    TEST_F(Library_Test, sum) {
+        auto [testGen, status] = createTestForFunction(test_c, 7);
```
Comment (ladisgin) [inline server/test/framework/Library_Test.cpp:38]: Add one more call `createTestForFunction(test_c, 7);` in same test

**Review by ladisgin — CHANGES_REQUESTED** [review]

File: integration-tests/c-example/CMakeLists.txt (hunk 1 above)
Comment (sava-cska) [reply integration-tests/c-example/CMakeLists.txt]: Done

File: server/src/building/Linker.cpp (hunk 2 above)
Comment (sava-cska) [reply server/src/building/Linker.cpp]: Done

File: server/test/framework/Library_Test.cpp (hunk 3 above)
Comment (sava-cska) [reply server/test/framework/Library_Test.cpp]: Done

**Review by ladisgin — APPROVED** [review]

**Pull request merged**